pkg/engine/armvariables.go

```diff
@@ -398,6 +391,7 @@ func getK8sMasterVars(cs *api.ContainerService) (map[string]interface{}, error)
 	}
 
 	masterVars["nsgID"] = "[resourceId('Microsoft.Network/networkSecurityGroups',variables('nsgName'))]"
+	masterVars["subnetName"] = cs.Properties.GetSubnetName()
```

Comment:
use `GetSubnetName()` function instead of embedding logic in armvariables file

Reply:
it's not clear to me that the existing GetSubnetName() returns the equivalent string responses as the above foo

Reply:
yeah I think you're right, looking into it... I assumed the existing function did the same thing but looks like it didn't

Reply:
I think the existing implementation depends upon ARM params usage patterns, so maybe that means it's a candidate for a future refactor phase.